pkg/abstractions/volume/http.go

```diff
@@ -121,14 +162,14 @@ func (g *volumeGroup) DownloadFile(ctx echo.Context) error {
 		return echo.NewHTTPError(http.StatusBadRequest, "Invalid volume path")
 	}
 
-	if f, err := g.gvs.getFileFd(
+	if path, err := g.gvs.getFilePath(
 		ctx.Request().Context(),
 		decodedVolumePath,
 		&workspace,
 	); err != nil {
 		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to download file %v", err))
 	} else {
-		return ctx.Stream(http.StatusOK, "application/octet-stream", f)
+		return ctx.File(path)
```

Comment:
I think this loads the entire file into memory which we don't really want. Can you double check? 

Also curious why Stream() doesn't work here.

Reply:
Stream doesnt provide file related information like content-type or content-length. The browser won't know how much it needs to download. 
I also think the File response should be optimized and use a reader. I can verify it though

Reply:
Right, but you can perform a best-effort detection. [This](https://github.com/beam-cloud/beta9/blob/main/pkg/abstractions/output/http.go#L39-L51) might help.

Reply:
I just tested this with a 5.25 gb file. Memory of the pod didnt change. In fact, execing into the pod shows that its not using significant memory

Reply:
I looked through the call stack 

`File() -> FileFS() -> fsFile() -> http.ServeContent() -> serveContent()`

And it doesn't look like it loads the entire file into memory. It does seek through the file to get the size of it. I'm not sure how juicefs processes this e.g. does it download the entire file to Seek or does it just get and use the metadata attributes from juicefs' redis.

Reply:
For what its worth, I asked ChatGPT and it says juicefs just uses metadata in redis when you os.Seek on a file, but I haven't checked juicefs myself.

Also - are you sure Stream doesn't work through the browser? I just downloaded an output (a text file) via the browser just fine. Are you seeing a specific issue with this? I can see maybe content type may not work since in outputs we're not using `mime.TypeByExtension()`.

Reply:
I think the biggest issue is that the browser won't know the content-length so it won't be able to get time estimates for the download as well as the progress
<img width="374" alt="image" src="https://github.com/beam-cloud/beta9/assets/91754185/d9f59057-e107-4ea2-ab4c-4dab1763816c">
<img width="553" alt="image" src="https://github.com/beam-cloud/beta9/assets/91754185/95aa241a-cbd2-429d-886c-0b4754e1ffdb">

vs using this
<img width="377" alt="image" src="https://github.com/beam-cloud/beta9/assets/91754185/b79680e6-5f16-41df-9bcb-01e8a5b3d566">
<img width="595" alt="image" src="https://github.com/beam-cloud/beta9/assets/91754185/eb51939a-f539-4fec-8bcb-950e371c3044">

Reply:
Hm I see. It still works, but from a UX perspective it's not the best.

Trust in ChatGPT? I'll leave this up to you to investigate or not.

Reply:
For the metadata, I was under the assumption that the redis stores the os.stat information

Reply:
The underlying function performs an os.Seek, not os.Stat. This is what I'm unsure of. ChatGPT says os.Seek still uses metadata.